### PR TITLE
Fix failing descriptor eval by pinning tiff library and fixing github actions

### LINKED ISSUE #30: Pillow Image import failing in descriptor eval script
On the production competition container image running in the code execution environment, the descriptor evaluation is failing:

```
conda run --no-capture-output -n condaenv python /opt/descriptor_eval.py --query_features query_descriptors.npz --ref_features reference_descriptors.npz --candidates_output full_rankings.csv
Traceback (most recent call last):
File "/opt/descriptor_eval.py", line 13, in <module>
from vsc.descriptor_eval_lib import evaluate_descriptor_track
File "/opt/vsc/descriptor_eval_lib.py", line 9, in <module>
from vsc.candidates import CandidateGeneration, MaxScoreAggregation
File "/opt/vsc/candidates.py", line 11, in <module>
from vsc.metrics import CandidatePair
File "/opt/vsc/metrics.py", line 14, in <module>
import matplotlib.pyplot as plt
File "/srv/conda/envs/condaenv/lib/python3.9/site-packages/matplotlib/__init__.py", line 113, in <module>
from . import _api, _version, cbook, _docstring, rcsetup
File "/srv/conda/envs/condaenv/lib/python3.9/site-packages/matplotlib/rcsetup.py", line 27, in <module>
from matplotlib.colors import Colormap, is_color_like
File "/srv/conda/envs/condaenv/lib/python3.9/site-packages/matplotlib/colors.py", line 51, in <module>
from PIL import Image
File "/srv/conda/envs/condaenv/lib/python3.9/site-packages/PIL/Image.py", line 114, in <module>
from . import _imaging as core
ImportError: libtiff.so.5: cannot open shared object file: No such file or directory
ERROR conda.cli.main_run:execute(47): `conda run python /opt/descriptor_eval.py --query_features query_descriptors.npz --ref_features reference_descriptors.npz --candidates_output full_rankings.csv` failed. (See above for error)
```

I have reproduced in the environment despite all local tests passing.

### LINKED ISSUE #32: GitHub actions pulls current version of image for testing
Desired behavior for automated container testing is for containers to be built and then have makefile tested. Currently, the makefile tests are running on the currently deployed image.

### PR DESCRIPTION
**Pull request by chrisjkuch**
This PR fixes failing descriptor eval by pinning `libtiff` libraries needed by PIL to v4.4.

This PR also fixes an issue with the previous makefile testing github action by directly testing the image after it is built in the runtime image by packing and submitting the quickstart solution.

Closes #30
Closes #32

### CODE REVIEW

**Review by jayqi — APPROVED** [review]

File: .github/workflows/build-images.yml [hunk 1]
```diff
@@ -44,6 +45,31 @@ jobs:
...
             -c "conda run --no-capture-output -n condaenv pytest tests/test_packages.py"
 
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.9"
+          cache: "pip"
+
+      - name: Install requirements
+        if: ${{ matrix.proc == 'cpu' }}
+        run: pip install -r requirements.txt
```
Comment (jayqi) [inline .github/workflows/build-images.yml:55]: Why do we need to setup Python?

File: .github/workflows/build-images.yml [hunk 2]
```diff
@@ -44,6 +45,31 @@ jobs:
...
+        if: ${{ matrix.proc == 'cpu' }}
+        run: |
+          make pack-quickstart
+
+      - name: Run submission in container
+        if: ${{ matrix.proc == 'cpu' }}
+        run: |
+          make CPU_OR_GPU=${{ matrix.proc }} \
+            SUBMISSION_IMAGE=$LOGIN_SERVER/$IMAGE:$SHA_TAG \
+            test-submission
```
Comment (jayqi) [inline .github/workflows/build-images.yml:71]: In the interest of good-enough, I don't think this is blocking. However, one thing to note is that the GPU job could finish running and publish the GPU image even if the CPU image fails its test. 

The way to solve this is to break the workflow up into sequential jobs, where publishing the image depends on the build and test for both images passing. There's some cleverness that's needed to cache the built image and not need to rebuild it in each stage, in order to make this happen.

**Review by jayqi — APPROVED** [review]
Looks good. Mainly one question about whether Python is needed.

File: .github/workflows/build-images.yml (hunk 1 above)
Comment (chrisjkuch) [reply .github/workflows/build-images.yml:55]: Generating the quickstart submission relies on numpy & pandas to generate random descriptors

File: .github/workflows/build-images.yml (hunk 2 above)
Comment (chrisjkuch) [reply .github/workflows/build-images.yml:71]: Great point. The protection against that for now is to only publish when merging into main, but this is something that we should explore further if we continue to see container issues.

**Pull request merged**